platforms/r3-corda/configuration/roles/delete/gitrepositories/tasks/main.yaml

```diff
@@ -0,0 +1,37 @@
+# List GitRepositories in namespace
+- name: List GitRepositories in namespace
+  k8s_info:
+    api_version: source.toolkit.fluxcd.io/v1beta2
+    kind: GitRepository
+    namespace: flux-{{ network.env.type }}
+  loop: "{{ network['organizations'] }}"
+  register: gitrepo_list
+
+# Patching GitRepository resources only if it exists
+- name: Patching GitRepository resources
+  k8s:
+    api_version: source.toolkit.fluxcd.io/v1beta2
+    kind: GitRepository
+    namespace: flux-{{ network.env.type }}
+    name: flux-{{ network.env.type }}
+    state: present
+    definition:
+      metadata:
+        finalizers: []
+  register: patch_result
+  loop: "{{ network['organizations'] }}"
+  when: ( gitrepo_list.results | json_query('[].resources') | flatten | length ) > 0
+
+# Deleting GitRepository resources only if it exists
+- name: Delete GitRepository resource
+  k8s:
+    api_version: source.toolkit.fluxcd.io/v1beta2
+    kind: GitRepository
+    namespace: flux-{{ network.env.type }}
+    name: flux-{{ network.env.type }}
+    state: absent
+  loop: "{{ network['organizations'] }}"
+  when:
+    - patch_result is succeeded
+    - ( gitrepo_list.results | json_query('[].resources') | flatten | length ) > 0
+    
```

Comment:
Always have a new line in files(no extra spaces, nothing, just press enter at the last line)